source/detail/environment.hpp

```diff
@@ -63,11 +65,26 @@
 
 /// <summary>
 /// Returns whether the C version `version` is supported by the current build configuration.
+/// Unfortunately, while __STDC_VERSION__ must be defined in C compilers, when using C++ compilers it is
+/// implementation-defined whether the __STDC_VERSION__ is defined or not. However, the C++ standard defines
+/// for each C++ version which C version it refers to. This can be found out by using the following list:
+/// C++98 -> C95
+/// C++11 -> C99
+/// C++17 -> C11
+/// C++20 -> C17
+/// C++26 -> C23
 /// </summary>
 /// <seealso cref="XLNT_C_11">
 /// <seealso cref="XLNT_C_17">
 /// <seealso cref="XLNT_C_23">
-#define XLNT_HAS_C_VERSION(version) (1/version == 1/version && XLNT_C_VERSION >= version)
+#define XLNT_HAS_C_VERSION(version) (1/version == 1/version && \
```

Comment:
It may be better/easier/cleaner to first define `XLNT_C_VERSION`. This would also have avoided the compile warning that you fixed in your next commit.

Should we fallback to deriving C version from C++ version if  `__STDC_VERSION__` is defined?

Reply:
> It may be better/easier/cleaner to first define `XLNT_C_VERSION`. This would also have avoided the compile warning that you fixed in your next commit.

I thought about that, but we would only define `XLNT_C_VERSION` as `__STDC_VERSION__`, which means that having an `XLNT_C_VERSION` wouldn't make much sense  (compared to `XLNT_CPP_VERSION` which exists because of MSVC). The thing is, while I wanted to use `defined(__STDC_VERSION__)` at first, it is actually simpler/shorter (but perhaps less clear) to simply use the definition, as it will evaluate to false if it's not defined. Of course, if you see any other advantage, let me know :wink: 

> Should we fallback to deriving C version from C++ version if `__STDC_VERSION__` is defined?

We could theoretically differentiate between C and C++ compilation by checking whether `__cplusplus` is defined. However, if defined, I would highly expect `__STDC_VERSION__` to always be the C version that the C++ standard refers to - anything else wouldn't make much sense. Unfortunately I did not test multiple compilers to check whether my assumption is true, so I can't really say what would be better.

Reply:
What I suggested was to combine `__STDC_VERSION__` and `XLNT_CPP_VERSION` into `XLNT_VERSION`, so this macro would just become `(1/version == 1/version && XLNT_C_VERSION >= version)`.

Reply:
Hmm, how should this work, considering that C and C++ have different version numbers? Can you provide an example?